RELEASE-NOTES-v1.2.md

```diff
@@ -0,0 +1,94 @@
+
+In the process of tackling these two issues, we addressed a number of other issues too (see "Other issues addressed" below) and added a monadic interface to the library so monadic pseudo-random number generators can be used interchangeably with `random`, see "API changes" below.
+
+This PR is the result of that effort. The changes are considerable. To signal this, we propose to release this as version 1.2 (the previous released version is 1.1, from 2014).
+
+However, the API changes are generally backwards-compatible, see "Compatibility" below.
+
+# Quality ([#25][issue-25])
+
+We created an [environment][quality-repo] for running statistical pseudo-random number generator tests, tested `random` v1.1 and `splitmix` using dieharder, TestU01, PractRand and other test suites and [recorded the results][quality-results].
+
+The results clearly show that the `split` operation in `random` v1.1 produces pseudo-random number generators which are correlated, corroborating [#25][issue-25]. The `split` operation in `splitmix` showed no weakness in our tests.
+
+As a result, we replaced the pseudo-random number generator implementation in `random` by the one provided by `splitmix`.
+
+# Performance ([#51][issue-51])
+
+@lehins' [performance analysis][analysis] has the data for `random` v1.1. It is slow, and using faster pseudo-random number generators via `random` v1.1 makes them slow.
+
+By switching to `splitmix` and improving the API, this PR speeds up pseudo-random number generation with `random` by one to three orders of magnitude, depending on the number type. See [Benchmarks][benchmarks] for details.
+
+# API changes
+
+The `Random` typeclass has conceptually been split into [`Uniform` and `UniformRange`][uniform-vs-uniformrange]. The `Random` typeclass is still included for backwards compatibility. `Uniform` is for types where it is possible to sample from the type's entire domain; `UniformRange` is for types where one can sample from a specified range.
+
+A new module [`System.Random.Monad`][random-monad] allows monadic pseudo-random number generators like `mwc-random` to be used via the new interface provided by `random`, see the docs for an [example of how this works][mwc-example].
+
+In addition, the module provides a convenient way to [run pure generators in monadic code][pure-gen].
+
+# Changes left out
+
+There were changes we considered and decided against including in this PR.
+
+Some pseudo-random number generators are splittable, others are not. A good way of communicating this is to have a separate typeclass, `Splittable`, say, which only splittable generators implement. After long discussions (see [this issue][split-issue] and [this PR][split-pr]), we decided against adding `Splittable`: the interface changes would either have been backwards-incompatible or too complex. For now, `split` stays part of the `RandomGen` typeclass. The new documentation suggests that [`split` should throw][split-docs] if the generator is not splittable.
```

Comment:
Is that the right Haskell term to use for `error`? If it says "throw an exception", is `error` what people think of?

Alternatives:
1. `split` should throw an exception
2. `split` should throw an imprecise exception (terminology from Arnaud's blog post, https://www.tweag.io/posts/2020-04-16-exceptions-in-haskell.html)
3. `split` should call `error`

Reply:
I've gone for "`split` should call `error`" for now. Not elegant, but clear.

Reply:
Technically it is called "bottom", but either `error`, `throw` or `undefined` will be exactly the same thing: an exception raised from pure code.

"Throwing an exception" on the other hand I think carries a bit more meaning, because it provides a reason for failure, which can only be recovered in `IO`, so I think "`split should call error`" is just fine